pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler_test.go

```diff
@@ -389,7 +393,7 @@ func TestCreateDefaultDeployment(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got := createRawDeployment(tt.args.objectMeta, tt.args.workerObjectMeta, tt.args.componentExt, tt.args.podSpec, tt.args.workerPodSpec)
+			got, _ := createRawDeployment(clientset, tt.args.objectMeta, tt.args.workerObjectMeta, tt.args.componentExt, tt.args.podSpec, tt.args.workerPodSpec)
```

Comment:
The function will always return an error here because during the function execution it won't be able to fetch the configmap to attach the oauth container. However the created deployment before encountering the oauth error will still be returned. So I think it is correct to ignore this error, right?

Reply:
In reality, you just proved me right.

> However the created deployment before encountering the oauth error will still be returned. 

That's not true. As part of your changes, you have `return nil, err` if the ConfigMap cannot be read. If `createRawDeployment` is always returning an error, I don't see any possibility that `got` has any value. So, these units are currently void.

The units are passing because in the for loop that follows `for i, deploy := range got`, it is tolerated that `got` is Nil. It simply won't do any iterations and no assertions are being done.

Reply:
I changed the code such that the `createRawDeployment` function will return a deployment list even if the configmap cannot be read. Verified by debugging the test that `got` is not empty everywhere

Reply:
@Jooho would this be fine for you?

I think Vedant's change is fine for non multi-node cases. However, for units involving multi-node, my impression is that the deployment list is going to be incomplete and can lead to units giving false positives.

I'll defer to @Jooho here. Minimally, I'd recommend changing the conditional to:
```
if len(got) != len(tt.expected) {
  t.Errorf("Got unexpected number of deployments")
}
```
to all instances.

Reply:
I don't think this is right. I think the problem is that ignoring err in test itself. And I added a comment, but I don't understand why an empty object is returned when an error occurs while retrieving configmap. It would be better to add it to test in case of an error.

Reply:
changed the logic flow such that the deploymentList is generated first and then oauth proxy is injected to the deployments in the list separately if the label is present